x/subscription/keeper/migrations.go

```diff
@@ -181,3 +184,55 @@ func (m Migrator) Migrate6to7(ctx sdk.Context) error {
 
 	return nil
 }
+
+// Migrate7to8 implements store migration from v7 to v8:
+// init new credit field
+func (m Migrator) Migrate7to8(ctx sdk.Context) error {
+	utils.LavaFormatDebug("migrate 7->8: subscriptions")
+
+	for _, index := range m.keeper.subsFS.GetAllEntryIndices(ctx) {
+		for _, block := range m.keeper.subsFS.GetAllEntryVersions(ctx, index) {
+			var s7 v7.Subscription
+			found := m.keeper.subsFS.FindEntry(ctx, index, block, &s7)
+			if !found {
+				utils.LavaFormatError("cannot migrate sub", fmt.Errorf("sub not found"),
+					utils.Attribute{Key: "index", Value: index},
+					utils.Attribute{Key: "block", Value: block},
+				)
+			}
+
+			var futureSub v8.FutureSubscription
+			if s7.FutureSubscription != nil {
+				futureSub = v8.FutureSubscription{
+					Creator:        s7.FutureSubscription.Creator,
+					PlanIndex:      s7.FutureSubscription.PlanIndex,
+					PlanBlock:      s7.FutureSubscription.PlanBlock,
+					DurationBought: s7.FutureSubscription.DurationBought,
+					Credit:         sdk.NewCoin(commontypes.TokenDenom, sdk.ZeroInt()),
+				}
+			}
+
+			s8 := v8.Subscription{
```

Comment:
i cool hack, you can use findentry to fill out the protobuf instead of copying it like this
found := m.keeper.subsFS.FindEntry(ctx, index, block, &s8)

Reply:
done. see https://github.com/lavanet/lava/pull/1141/commits/de4e3b29169abd1849c70732adc5469d5c6284a2